src/main.rs

```diff
@@ -71,6 +86,39 @@ fn main() {
     std::io::stdin().read_line(&mut String::new()).unwrap();
 }
 
+
+fn update() -> Result<(), Box<dyn std::error::Error>> {
+    info!("checking for updates");
+
+    let status = self_update::backends::github::Update::configure()
+        .repo_owner("IceDynamix")
+        .repo_name("reliquary-archiver")
+        .bin_name("reliquary-archiver")
+        .target("x64")
+        .show_download_progress(true)
+        .show_output(false)
+        .current_version(cargo_crate_version!())
+        .build()?
+        .update()?;
+
+    if status.updated() {
+        info!("updated to {}", status.version());
+
+        let current_exe = env::current_exe();
+        let mut command = Command::new(current_exe?);
+        command.args(env::args().skip(1)).env("NO_SELF_UPDATE", "");
```

Comment:
Probably a better idea to set a non-empty string for the env var as that is more obvious and less error-prone.
```suggestion
        command.args(env::args().skip(1)).env("NO_SELF_UPDATE", "1");
```